src/components/DelayUnmount.tsx

```diff
@@ -0,0 +1,36 @@
+/*!
+ * The contents of this file are subject to the Common Public Attribution License Version 1.0 (the “License”); you may not use this file except in compliance with the License. You may obtain a copy of the License at https://mpapp-public.gitlab.io/manuscripts-frontend/LICENSE. The License is based on the Mozilla Public License Version 1.1 but Sections 14 and 15 have been added to cover use of software over a computer network and provide for limited attribution for the Original Developer. In addition, Exhibit A has been modified to be consistent with Exhibit B.
+ *
+ * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
+ *
+ * The Original Code is manuscripts-frontend.
+ *
+ * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
+ *
+ * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
+ */
+import React, { useEffect, useState } from 'react'
+
+export const DelayUnmount: React.FC<{
+  isVisible: boolean
+  children: React.ReactNode
+  delay?: number
+}> = ({ isVisible, delay, children }) => {
+  const [shouldRender, setShouldRender] = useState(true)
+
+  useEffect(() => {
+    let timeoutId: number
+    if (!isVisible && shouldRender) {
+      // @ts-ignore
+      timeoutId = setTimeout(() => {
+        setShouldRender(false)
+      }, delay || 200)
+    }
+    if (isVisible && !shouldRender) {
+      setShouldRender(true)
+    }
+    return () => clearTimeout(timeoutId)
+  }, [isVisible, delay, shouldRender])
+
+  return shouldRender ? <>{children}</> : null
+}
```

Comment:
There is a lag when writing in the search text input, some of the text in input field gets cut so you will need to input each character slowly.

What about rendering `AdvancedSearch.tsx` from the plugin view of `search-replace.ts` and just debounce text input as we do for other elements

Reply:
I did observe lag only with devtools enabled. Does it happen to you with devtools disabled too?

Reply:
The lag that I observed (with inspector on), is due to a large amount of decorations being rendered at the same time. I don't think rendering them with debounce will help if you need to select 1 letter. Although that would definitely help when you need to type a lot of text. This is however not related to DelayUnmount.

The reason why I didn't want to use pluginView for that, is because this view is relatively complex and there is no place for it in the body-editor because it's not related to any node so there is nowhere to append it except to the rootNode. I didn't find that fitting to create and manage rendering of a complex react component from the plugin view. And also all generic control that we have, we normally render in the article editor.

Reply:
I see the delay now, I'll have a look

Reply:
I added debounce. Seems like it fixed the problem.